cmd/daemon/main.go

```diff
@@ -414,24 +425,46 @@ func (d *daemon) make(c *cli.Context) error {
 		}
 	}()
 
-	log.Infof("started swapd with basepath %s",
-		cfg.Basepath,
-	)
+	log.Infof("started swapd with data-dir %s", cfg.DataDir)
 	return nil
 }
 
-func newBackend(ctx context.Context, c *cli.Context, env common.Environment, cfg common.Config,
-	chainID int64, devXMRMaker bool, sm swap.Manager, net net.Host) (backend.Backend, error) {
+func errFlagsMutuallyExclusive(flag1, flag2 string) error {
+	return fmt.Errorf("flags %q and %q are mutually exclusive", flag1, flag2)
+}
+
+func errFlagRequired(flag string) error {
+	return fmt.Errorf("required flag %q not specified", flag)
+}
+
+func newBackend(
+	ctx context.Context,
+	c *cli.Context,
+	env common.Environment,
+	cfg common.Config,
+	devXMRMaker bool,
+	devXMRTaker bool,
```

Comment:
We are passing both `devXMRMaker` and `devXMRTaker` now, as it is possible to be in a dev environment and be neither one of these. Before, code was assuming that you must be `devXMRTaker` if you are not `devXMRMaker` in a dev environment. Longer term, I'm thinking we can eliminate these flags and just use config files with the same values set.